pypeit/images/rawimage.py

```diff
@@ -19,399 +19,701 @@
+            msgs.info(f'Estimated readnoise of amplifier {amp+1} = {self.ronoise[amp]:.3f} e-')
+
+    def build_rn2img(self, units='e-', digitization=True):
         """
-        Generate the model read noise squared image
+        Generate the model detector variance image (:attr:`rn2img`).
 
-        Currently only used by ScienceImage.
+        This is primarily a wrapper for :func:`~pypeit.core.procimg.rn2_frame`.
 
-        Wrapper to procimg.rn_frame
+        Args:
+            units (:obj:`str`, optional):
+                Units for the output variance.  Options are ``'e-'`` for
+                variance in square electrons (counts) or ``'ADU'`` for square
+                ADU.
 
         Returns:
-            `numpy.ndarray`_: Copy of the read noise squared image
-
+            `numpy.ndarray`_: Readnoise variance image.
         """
-        # Check if we need to determine the read noise directly from the overscan region from any amplifier
-        numamps = len(self.ronoise)
-        for amp in range(numamps):
-            if self.ronoise[amp] <= 0.0:
-                biaspix = self.rawimage[self.oscansec_img==amp+1] * self.detector['gain'][amp]
-                _, _, stddev = stats.sigma_clipped_stats(biaspix, sigma=5)
-                self.ronoise[amp] = stddev
-                msgs.info("Read noise of amplifier {0:d} = {1:.3f} e-".format(amp+1, self.ronoise[amp]))
-
-        # Build it
-        self.rn2img = procimg.rn_frame(self.datasec_img,
-                                       self.detector['gain'],
-                                       self.ronoise)
-        # Return
-        return self.rn2img.copy()
-
-    def process(self, par, bpm=bpm, flatimages=None, bias=None,
-                slits=None, debug=False, dark=None):
+        if not np.all(self.ronoise > 0):
+            # TODO: Consider just calling estimate_readnoise here...
+            msgs.error('Some readnoise values <=0; first call estimate_readnoise.')
+        # Compute and return the readnoise variance image 
+        return procimg.rn2_frame(self.datasec_img, self.detector['gain'], self.ronoise,
+                                 units=units, digitization=digitization)
+
+    def process(self, par, bpm=None, flatimages=None, bias=None, slits=None, debug=False,
+                dark=None):
         """
-        Process the image
+        Process the image.
+
+        See further discussion of :ref:`image_proc` in ``PypeIt``.
+
+        .. note::
+
+            The processing step order is currently 'frozen' as is.  However, in
+            the future, we may choose to allow optional ordering.
+
+        The processing steps used (depending on the parameter toggling in
+        :attr:`par`), in the order they will be applied are:
+
+            #. :func:`subtract_pattern`: Analyze and subtract the pattern noise
+               from the image.
+
+            #. :func:`build_rn2img`: Construct the detector variance image,
```

Comment:
I would not call this a detector variance image. That is not defined and is likely to get confused with the variance image, which is the full noise model of an image. The rn2img is the contribution to the noise coming only from read noise,  digitisation noise, and dark current. I think that is what you mean by "detector variance image", but let's not create a new term for it. Just call it rn2 img since that is how it is used in the code.

Reply:
Changed to "readnoise variance image"